doc/dev/model.md

```diff
@@ -0,0 +1,78 @@
+# Overview
+
+This document provides an introduction to the msrest model and dpg model architectures.
+
+## msrest Model
```

Comment:
Same feedback as before, for customers we shouldn't mention msrest. It's not even a dependency anymore for a while.